CMakeLists.txt

```diff
@@ -143,6 +143,11 @@ endif (UNIX)
 
 # many are core-specific
 
+# whether this build of DynamoRIO is meant for distribution, in which case
+# glibc should not have any too-recent imports. See core/CMake_readelf.cmake
+# for more information.
+option(BUILD_PACKAGE "build DynamoRIO for packaging purposes by performing glibc checks" ON)
```

Comment:
Given https://groups.google.com/g/dynamorio-users/c/Y2cUFCgESHA/m/GLq46qGnBgAJ and others hitting this I'm thinking this should be OFF by default and turned on by our package.cmake.  The logic is that a user obtaining our sources should be able to build for the local machine without having to say "-DBUILD_PACKAGE=OFF"; put the onus on those building published packages to set a non-default value as those users/scripts are already setting various things -- keep the new user "just build this locally" simple.  (I don't think breaking the current default is a significant concern -- maybe add a note to the changelog in api/docs/release.dox.)